core/src/main/kotlin/org/evomaster/core/problem/rest/resource/RestResourceNode.kt

```diff
@@ -0,0 +1,818 @@
+package org.evomaster.core.problem.rest.resource
+
+import org.evomaster.core.database.DbAction
+import org.evomaster.core.problem.rest.HttpVerb
+import org.evomaster.core.problem.rest.RestAction
+import org.evomaster.core.problem.rest.RestCallAction
+import org.evomaster.core.problem.rest.RestPath
+import org.evomaster.core.problem.rest.param.BodyParam
+import org.evomaster.core.problem.rest.param.Param
+import org.evomaster.core.problem.rest.param.PathParam
+import org.evomaster.core.problem.rest.resource.dependency.*
+import org.evomaster.core.problem.rest.util.ParamUtil
+import org.evomaster.core.problem.rest.util.ParserUtil
+import org.evomaster.core.problem.rest.util.RestResourceTemplateHandler
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.gene.ObjectGene
+import org.evomaster.core.search.service.Randomness
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * @property path resource path
+ * @property actions actions under the resource, with references of tables
+ * @property initMode configurable option to init resource with additional info, e.g., related tables
+ */
+class RestResourceNode(
+        val path : RestPath,
+        val actions: MutableList<RestAction> = mutableListOf(),
+        val initMode : InitMode
+) {
+
+    companion object {
+        private const val PROB_EXTRA_PATCH = 0.8
+        val log: Logger = LoggerFactory.getLogger(RestResourceNode::class.java)
+    }
+
+    private val tokens : MutableMap<String, PathRToken> = mutableMapOf()
+
+    private val segments : MutableMap<Boolean, List<String>> = mutableMapOf()
```

Comment:
a map where the key is a boolean??? (which can only have 2 values)
seems bit weird